api/log/log.go

```diff
@@ -2,13 +2,31 @@
 package log
 
 import (
+	"context"
 	"fmt"
+	"log/slog"
 	"net/http"
 	"os"
 
 	"github.com/pkg/errors"
 )
 
+// ErrorKey is the logging attribute key for error values.
+const ErrorKey = "error"
+
+type loggerKey struct{}
+
+// NewContext creates a new context with the given slog.Logger.
+func NewContext(ctx context.Context, logger *slog.Logger) context.Context {
```

Comment:
I think that instead of attaching a `slog.Logger` reference to the `context.Context`, we may benefit more by attaching a `func`. This way we can inject this concern in more elastic ways in the future.

Thoughts?

```go
type HandlerErrorLogger func(http.ResponseWriter, *http.Request, error)

func (fn HandlerErrorLogger) call(w http.ResponseWriter, r *http.Request, err error) {
	if fn == nil {
		return // noop or use a sensible default (print to stderr via slog, etc)
	}

	fn(w, r, err)
}

func WithHandlerErrorLogger(ctx context.Context, fn HandlerErrorLogger) context.Context {
	// ...
}

func HandlerErrorLoggerFromContext(ctx context.Context) HandlerErrorLogger {
	// ...
}

func Error(rw http.ResponseWriter, r *http.Request, err error) {
	HandlerErrorLoggerFromContext(r.Context()).call(rw, r, err)
	// ...
}
```

Thoughts?

Reply:
Fixed with c0b7c33a